toolkit/tools/imagegen/configuration/systemconfig.go

```diff
@@ -66,6 +67,34 @@ func (s *SystemConfig) GetMountpointPartitionSetting(mountPoint string) (partiti
 	return FindMountpointPartitionSetting(s.PartitionSettings, mountPoint)
 }
 
+func (s *SystemConfig) validateUsersAndGroups() (err error) {
+	for _, user := range s.Users {
+		groupMatchFunc := func(groupName interface{}, groupObj interface{}) bool {
+			return groupName == groupObj.(Group).Name
+		}
+
+		if user.PrimaryGroup != "" {
+			if !sliceutils.Contains(s.Groups, user.PrimaryGroup, groupMatchFunc) {
+				// Unclear how to validate these while allowing users to be added to existing system groups. If we
+				// define a group in the config that already exists it will cause errors.
+				// Maybe we can scrape /etc/group and /etc/passwd from filesystem.sepc to validate these?
+				logger.Log.Warnf("Primary group (%s) for user (%s) not defined", user.PrimaryGroup, user.Name)
+			}
+		}
+
+		for _, group := range user.SecondaryGroups {
+			if !sliceutils.Contains(s.Groups, group, groupMatchFunc) {
+				// Unclear how to validate these while allowing users to be added to existing system groups. If we
+				// define a group in the config that already exists it will cause errors.
+				// Maybe we can scrape /etc/group and /etc/passwd from filesystem.sepc to validate these?
+				logger.Log.Warnf("secondary group (%s) for user (%s) not defined", group, user.Name)
```

Comment:
nit: `Secondary`. (Logs start with an uppercase letter. Errors start with a lowercase letter.)

Reply:
Also add a period at the end, please.